compiler/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -833,6 +852,35 @@ impl<'interner> TypeChecker<'interner> {
                     }
                 }
             }
+            Type::NamedGeneric(_, _) => {
+                let func_meta = self.interner.function_meta(&self.current_function.unwrap());
+
+                for constraint in func_meta.trait_constraints {
+                    if let Some(trait_id) = constraint.trait_id {
+                        // TODO(#2568): == on types is sketchy, since Field != TypeVar::Bound(Field)
+                        // unify() is sketchier here though, since it may accidentally commit typebindings.
+                        // this works for now, but likely needs to be revisited when we implement generic traits
+                        if *object_type == constraint.typ {
```

Comment:
You're right, I was mostly worried about the `Trait::self_type_typevar` that I added in the previous PR (https://github.com/noir-lang/noir/blob/master/compiler/noirc_frontend/src/hir/def_collector/dc_crate.rs#L747), but that's impossible to reach here, as it would be instantiated during method signature type checking.

I've removed this comment: https://github.com/noir-lang/noir/pull/2717/commits/e12234853bf747664a459d035a4d176a76171803